src/basilisp/core.lpy

```diff
@@ -4486,7 +4494,10 @@
                  (= (:eof opts) :eofthrow)
                  (:features opts)
                  (not= :preserve (:read-cond opts))
-                 *default-data-reader-fn*)))))
+                 *default-data-reader-fn*
+                 **
+                 :line_num (:line_num opts)
+                 :column_num (:column_num opts))))))
```

Comment:
```suggestion
                 :line-num (:init-line opts)
                 :column-num (:init-col opts))))))
```

Two minor points:
1. It's acceptable (and preferred) to just use kebab-case for Python kwargs which are simple snake case. The compiler munges them correctly into the corresponding Python name.
2. I'm suggesting changing the name around slightly (throughout the entire PR) to something like `:init-line` and `:init-col` to indicate its true meaning (the initial location for the reader).

Reply:
Updated naming throughout.

Oops, just realized I used `init-column` instead of `init-col`. Let me know if this affects consistency.